src/main/java/minijava/parser/Parser.java

```diff
@@ -230,7 +238,8 @@ private void parseBlock() {
   /** BlockStatement -> Statement | LocalVariableDeclarationStatement */
   private void parseBlockStatement() {
     if (currentToken.isOneOf(INT, BOOLEAN, VOID)
-        || matchCurrentAndLookAhead(IDENT, LBRACKET, RBRACKET)) {
+        || matchCurrentAndLookAhead(IDENT, LBRACKET, RBRACKET)
+        || matchCurrentAndLookAhead(IDENT, IDENT)) {
```

Comment:
Obviously this wouldn't have matched stuff like `String s = "";`, because `String` is an `IDENT`, but `s` is also an `IDENT` and not an `LBRACKET`.